src/nodes/lighting/LightsNode.js

```diff
@@ -114,27 +115,31 @@ class LightsNode extends Node {
 	 */
 	customCacheKey() {
 
-		const hashData = [];
 		const lights = this._lights;
 
 		for ( let i = 0; i < lights.length; i ++ ) {
 
 			const light = lights[ i ];
 
-			hashData.push( light.id );
+			_hashData.push( light.id );
+			_hashData.push( light.castShadow ? 1 : 0 );
 
 			if ( light.isSpotLight === true ) {
 
 				const hashMap = ( light.map !== null ) ? light.map.id : - 1;
 				const hashColorNode = ( light.colorNode ) ? light.colorNode.getCacheKey() : - 1;
 
-				hashData.push( hashMap, hashColorNode );
+				_hashData.push( hashMap, hashColorNode );
 
 			}
 
 		}
 
-		return hashArray( hashData );
+		const cacheKey = hashArray( _hashData );
+
+		_hashData.length = 0;
```

Comment:
Do we have any source that `.length=0` is better than other alternatives?

I didn't do an extensive beachmark, but I had done some tests between creating a new array and using `.length=0` and creating a new array worked better. It seems that both force the GC, using null values ​​after use was what worked best for small arrays.

Reply:
I've tried to search the for the original issue/PR where we discussed this once but did not find it. It's many years ago but at that time, setting a zero length was the fastest way to clear an array. And it does no create a new array object  like the previous approach which invalidates the previous object.

I'm surprised to hear that `.length=0` triggers the GC. The array itself is just empty and the reference to it is not set to `null` or cleared.

Reply:
We can revisit the array management at any point again. Merging for now so the toggling issue is fixed on dev.